clients/src/main/java/org/apache/kafka/clients/NetworkClientUtils.java

```diff
@@ -56,20 +56,19 @@ public static boolean awaitReady(KafkaClient client, Node node, Time time, long
         long startTime = time.milliseconds();
         long expiryTime = startTime + timeoutMs;
 
-        if (!isReady(client, node, startTime))
-            client.ready(node, startTime);
+        if (isReady(client, node, startTime) ||  client.ready(node, startTime))
```

Comment:
I was also trying to make the implementation as close as possible to what we had in `NetworkClientBlockingOps` since I know the details have been a bit tricky to get right. Also, I think "immediate connections" have actually popped up as an issue in `NetworkClient` which we've had to add code to ensure we handle properly.